tests/foreman/api/test_contentview.py

```diff
@@ -750,6 +751,98 @@ def test_composite_content_view_with_same_repos(self, module_org, target_sat):
         comp_content_view_info = comp_content_view.version[0].read()
         assert comp_content_view_info.package_count == 36
 
+    @pytest.mark.stream
+    @pytest.mark.tier2
+    def test_ccv_audit_scenarios(self, module_org, target_sat):
+        """Check for various scenarios where a composite content view or it's component
+        content views needs_publish flags should be set to true and that they properly
+        get set and unset
+
+        :id: cdd94ab8-da31-40ac-ab81-02472517e9bf
+
+        :expectedresults: When appropriate, a ccv and it's cvs needs_publish flags get
+            set or unset
+
+        :CaseLevel: Integration
+
+        :CaseImportance: High
+        """
+        composite_cv = target_sat.api.ContentView(composite=True).create()
+        # Needs_publish is set to True on creation
+        assert composite_cv.read().needs_publish
+        composite_cv.publish()
+        assert not composite_cv.read().needs_publish
+        # Add some content_views to the composite view
+        self.add_content_views_to_composite(composite_cv, module_org, 2)
+        # Needs_publish should be set to True when a component CV is added/removed
+        assert composite_cv.read().needs_publish
+        composite_cv.publish()
+        assert not composite_cv.read().needs_publish
+        component_cvs = composite_cv.read().component
+        # Remove a component cv, should need publish now
+        component_cvs.pop(1)
+        composite_cv.component = component_cvs
+        composite_cv = composite_cv.update(['component'])
+        assert composite_cv.read().needs_publish
+        composite_cv.publish()
+        assert not composite_cv.read().needs_publish
+        # add a CV that has `latest` set to true
+        new_cv = target_sat.api.ContentView().create()
+        new_cv.publish()
+        composite_cv.content_view_component[0].add(
+            data={"components": [{"content_view_id": new_cv.id, "latest": "true"}]}
+        )
+        assert composite_cv.read().needs_publish
+        composite_cv.publish()
+        assert not composite_cv.read().needs_publish
+        # a new version of a component cv that has "latest" - needs publish
+        new_cv.publish()
+        assert composite_cv.read().needs_publish
+        composite_cv.publish()
+        assert not composite_cv.read().needs_publish
+        # a component CV was changed to "always update" when ccv has old version - needs publish
+        # update composite_cv after changes
+        composite_cv = composite_cv.read()
+        # get the first CV that was added, which has 1 version
+        old_component = composite_cv.content_view_component[0].read()
```

Comment:
It works this way in practice, although to be completely sure I could save some information and fetch that - this was just a convenient way to fetch it as well. 

It does matter though, since the idea with this portion is to change a Component CV that has an old version included in the CCV to set `latest` to true, requiring a change